Samples/BackgroundBlur/BackgroundBlur/SegmentModel.cpp

```diff
@@ -0,0 +1,285 @@
+#include "SegmentModel.h"
+
+#include <iostream>
+#include <filesystem>
+
+using winrt::Windows::Foundation::PropertyValue;
+using winrt::hstring;
+using namespace winrt;
+using namespace Windows::Foundation::Collections;
+
+enum OnnxDataType : long {
+	ONNX_UNDEFINED = 0,
+	// Basic types.
+	ONNX_FLOAT = 1,
+	ONNX_UINT8 = 2,
+	ONNX_INT8 = 3,
+	ONNX_UINT16 = 4,
+	ONNX_INT16 = 5,
+	ONNX_INT32 = 6,
+	ONNX_INT64 = 7,
+	ONNX_STRING = 8,
+	ONNX_BOOL = 9,
+
+	// IEEE754 half-precision floating-point format (16 bits wide).
+	// This format has 1 sign bit, 5 exponent bits, and 10 mantissa bits.
+	ONNX_FLOAT16 = 10,
+
+	ONNX_DOUBLE = 11,
+	ONNX_UINT32 = 12,
+	ONNX_UINT64 = 13,
+	ONNX_COMPLEX64 = 14,     // complex with float32 real and imaginary components
+	ONNX_COMPLEX128 = 15,    // complex with float64 real and imaginary components
+
+	// Non-IEEE floating-point format based on IEEE754 single-precision
+	// floating-point number truncated to 16 bits.
+	// This format has 1 sign bit, 8 exponent bits, and 7 mantissa bits.
+	ONNX_BFLOAT16 = 16,
+}OnnxDataType;
+
+
+int g_scale = 5;
+auto outputBindProperties = PropertySet();
+
+/****	Style transfer model	****/
+void StyleTransfer::SetModels(int w, int h)
+{
+	// TODO: Use w/h or use the 720x720 of the mode
+	SetImageSize(720, 720); // SIze model input sizes fixed to 720x720
+	m_session = CreateLearningModelSession(GetModel());
+	m_binding = LearningModelBinding(m_session);
+}
+void StyleTransfer::Run(IDirect3DSurface src, IDirect3DSurface dest)
+{
+	m_bSyncStarted = TRUE;
+
+	assert(m_session.Device().AdapterId() == m_highPerfAdapter);
+	VideoFrame inVideoFrame = VideoFrame::CreateWithDirect3D11Surface(src);
+	VideoFrame outVideoFrame = VideoFrame::CreateWithDirect3D11Surface(dest);
+	SetVideoFrames(inVideoFrame, outVideoFrame);
+
+	hstring inputName = m_session.Model().InputFeatures().GetAt(0).Name();
+	m_binding.Bind(inputName, m_inputVideoFrame);
+	hstring outputName = m_session.Model().OutputFeatures().GetAt(0).Name();
+
+	auto outputBindProperties = PropertySet();
+	outputBindProperties.Insert(L"DisableTensorCpuSync", PropertyValue::CreateBoolean(true));
+
+	m_binding.Bind(outputName, m_outputVideoFrame, outputBindProperties); // TODO: See if can bind videoframe from MFT
+	auto results = m_session.Evaluate(m_binding, L"");
+
+	m_outputVideoFrame.CopyToAsync(outVideoFrame).get();
+
+	m_bSyncStarted = FALSE;
+}
+LearningModel StyleTransfer::GetModel()
+{
+	auto rel = std::filesystem::current_path();
+	rel.append("Assets\\mosaic.onnx");
+	return LearningModel::LoadFromFilePath(rel + L"");
```

Comment:
why do you add L""? is that to cast the string to a different type? seems like an odd way to do that.